python/lsst/ts/tcpip/base_client_or_server.py

```diff
@@ -332,6 +333,32 @@ async def read_into(self, struct: ctypes.Structure) -> None:
             await self._close_client()
             raise
 
+    async def read_str(self) -> str:
+        """Read data ending in `TERMINATOR`.
+
+        Before returning the data, it is decoded into an UTF-8 string.
+
+        Returns
+        -------
+        str:
+            Data decoded into a string.
+        """
+        data = await self.readuntil(TERMINATOR)
+        return data.decode(encoding="utf-8", errors="strict")
+
+    async def read_json(self) -> str:
+        """Read JSON data.
+
+        Before returning the data, it is decoded using JSON.
+
+        Returns
+        -------
+        str:
+            Data decoded into a string.
+        """
+        data = await self.read_str()
+        return json.loads(data)
```

Comment:
Is the json exception clear (i.e. does it include the data that it failed to parse)? If not, please consider a try/catch to raise a more informative error.